services/storage/tests/unit/test_handlers_files.py

```diff
@@ -0,0 +1,504 @@
+        file_id=simcore_file_id,
+        expected_entry_exists=True,
+        expected_file_size=-1,
+        expected_upload_expiration_date=True,
+    )
+
+    # abort file upload
+    abort_url = (
+        client.app.router["abort_upload_file"]
+        .url_for(
+            location_id=f"{location_id}",
+            file_id=urllib.parse.quote(simcore_file_id, safe=""),
+        )
+        .with_query(user_id=user_id)
+    )
+    response = await client.post(f"{abort_url}")
+    await assert_status(response, web.HTTPNoContent)
+
+    # the DB shall be cleaned up
+    await assert_file_meta_data_in_db(
+        aiopg_engine,
+        file_id=simcore_file_id,
+        expected_entry_exists=False,
+        expected_file_size=None,
+        expected_upload_expiration_date=None,
+    )
+
+
+@pytest.mark.parametrize(
+    "link_type, file_size",
+    [
+        (LinkType.PRESIGNED, parse_obj_as(ByteSize, "10Mib")),
+        (LinkType.PRESIGNED, parse_obj_as(ByteSize, "1000Mib")),
+        (LinkType.S3, parse_obj_as(ByteSize, "10Mib")),
+        (LinkType.S3, parse_obj_as(ByteSize, "1000Mib")),
+    ],
+)
+async def test_upload_same_file_uuid_aborts_previous_upload(
+    aiopg_engine: Engine,
+    client: TestClient,
+    storage_s3_client: StorageS3Client,
+    storage_s3_bucket: S3BucketName,
+    simcore_file_id: SimcoreS3FileID,
+    link_type: LinkType,
+    file_size: ByteSize,
+    create_upload_file_link: Callable[..., Awaitable[AnyUrl]],
+):
+    assert client.app
+    # create upload file link
+    file_upload_link = await create_upload_file_link(
+        simcore_file_id, link_type=link_type.value.lower(), file_size=file_size
+    )
+    # we shall have an entry in the db, waiting for upload
+    await assert_file_meta_data_in_db(
+        aiopg_engine,
+        file_id=simcore_file_id,
+        expected_entry_exists=True,
+        expected_file_size=-1,
+        expected_upload_expiration_date=True,
+    )
+
+    await asyncio.sleep(1)
+    # now we create a new upload
+    # we should abort the previous upload to prevent unwanted costs
+    new_file_upload_link = await create_upload_file_link(
+        simcore_file_id, link_type=link_type.value.lower(), file_size=file_size
+    )
+
+    if link_type == LinkType.PRESIGNED:
+        assert file_upload_link != new_file_upload_link
+    else:
+        assert file_upload_link == new_file_upload_link
+    # we shall have an entry in the db, waiting for upload
+    await assert_file_meta_data_in_db(
+        aiopg_engine,
+        file_id=simcore_file_id,
+        expected_entry_exists=True,
+        expected_file_size=-1,
+        expected_upload_expiration_date=True,
+    )
+
+
+@pytest.mark.parametrize(
+    "file_name",
+    [
+        "some file name with spaces and extension.txt",
+        "some name with special characters -_ü!öäàé++3245",
+    ],
+)
+@pytest.mark.parametrize(
+    "file_size",
+    [
+        pytest.param(parse_obj_as(ByteSize, "1Mib"), id="7Mib"),
```

Comment:
ok I added a fct for human readable parametrizations, this will fix this once and for all by using code such as:
```python
@pytest.mark.parametrize(
    "file_size, expected_num_chunks, expected_chunk_size",
    [
        (parse_obj_as(ByteSize, "5Mib"), 1, parse_obj_as(ByteSize, "10Mib")),
        (parse_obj_as(ByteSize, "10Mib"), 1, parse_obj_as(ByteSize, "10Mib")),
        (parse_obj_as(ByteSize, "20Mib"), 2, parse_obj_as(ByteSize, "10Mib")),
        (parse_obj_as(ByteSize, "50Mib"), 5, parse_obj_as(ByteSize, "10Mib")),
        (parse_obj_as(ByteSize, "150Mib"), 15, parse_obj_as(ByteSize, "10Mib")),
        (parse_obj_as(ByteSize, "550Mib"), 55, parse_obj_as(ByteSize, "10Mib")),
        (parse_obj_as(ByteSize, "560Gib"), 5735, parse_obj_as(ByteSize, "100Mib")),
        (parse_obj_as(ByteSize, "5Tib"), 8739, parse_obj_as(ByteSize, "600Mib")),
        (parse_obj_as(ByteSize, "15Tib"), 7680, parse_obj_as(ByteSize, "2Gib")),
    ],
    ids=byte_size_ids,
)
```

the function ```byte_size_ids``` is now in pytest-simcore in the helpers